### ISSUE TITLE
Add phase image as input channel to improve segmentation performance

### ISSUE THREAD

**Issue by jcohenadad** [ivadomed/model_seg_sc-gm-lesion_human_ms_exvivo_t2star#18]
**Description**
Currently, the model uses only the magnitude image for input. Incorporating the phase image as an additional input channel may provide complementary information, particularly near tissue boundaries or susceptibility-affected regions, potentially improving segmentation accuracy.

**Proposed Change**

* Modify the data loading pipeline to support two-channel input: magnitude + phase.
* Retrain model and compare with magnitude-only baseline.

**Rationale**
Preliminary observations suggest that phase contrast enhances anatomical delineation in our target regions (e.g., GM/WM interface, lesions), which could lead to improved pseudo-Dice scores.

**Impact**

* Minimal code change in preprocessing and model input layers.
* May require re-tuning of normalization and training schedule.

**Comment by jcohenadad** [comment]
Using prototype code in [721ef04](https://github.com/ivadomed/model_seg_sc-gm-lesion_human_ms_exvivo_t2star/commit/721ef0414169a760cb1bbf5faae07366aa8e0a4a):

~~~
python extract_slices.py --path-data ${PATH_DATA} --label-folder derivatives/labels_combined-sc-gm --labels combined --path-out ${PATH_PROCESSED}/data_slice --adjacent-slices 1 --use-phase
python convert_bids_to_nnunet.py --path-data ${PATH_PROCESSED}/data_slice --label-json ${PATH_DATA}/derivatives/labels_combined-sc-gm/sub-01/anat/sub-01_part-mag_chunk-02_T2starw_label-combined_classes.json --tasknumber 502 --adjacent-slices 1 --use-phase
~~~

Here is the dataset.json:
```json
{
    "channel_names": {
        "0": "T2starw_offset0_mag",
        "1": "T2starw_offset0_phase",
        "2": "T2starw_offset-1_mag",
        "3": "T2starw_offset-1_phase",
        "4": "T2starw_offset1_mag",
        "5": "T2starw_offset1_phase"
    },
    "labels": {
        "background": 0,
        "SC_without_GM": 1,
        "SC_GM": 2
    },
    "numTraining": 9,
    "file_ending": ".nii.gz",
    "licence": "CC-BY 4.0",
    "converted_by": "Please enter your name, especially when sharing datasets with others in a common infrastructure!",
    "name": "Segmentation",
    "reference": "Julien Cohen-Adad, Postmortem SC MRI",
    "release": "1.0.0",
    "overwrite_image_reader_writer": "NibabelIOWithReorient"
}
```

**Comment by jcohenadad** [comment]
Below comparison with/without the phase, using 3 adjacent slices:

![Image](https://github.com/user-attachments/assets/0f3ce439-e397-4635-bd16-8c91e50c0e0f)

Using the phase yields promising results (esp. because the number of epochs for the phase model was 230 vs. 520 without the phase). Further investigations are required.

**Comment by Luugaaa** [comment]
# Context 

We believe the phase carries meaningful information not necessarily present (or at least not obvious) in the magnitude. However, we know that the phase can also carry artifacts that can lead to mis-segmentation. For this reason we decided to run an investigation on the impact of giving the model both the magnitude and the phase.

# Method

Until now we gave the model a 2-channels image, the first channel being the magnitude and the second channel being the phase. 

For the purpose of the experiment we decided to give the model only the magnitude for comparison. Instead of giving a 1-channel image with the magnitude only, we replaced the phase in the second channel with the content of the first channel, giving the model the same magnitude image twice. This has two main advantages : 1) it keeps the modifications minimal 2) it guaranties the input format is the same for a better comparison. 

For the preprocessing we only used the otsu thresholding as decided in #24 

# Results 

Note: The following results are dice score averaged over 5 folds on the validation set of the "full" 12 subjects dataset. 

The colors are : 
- **gray:** models trained with 2 mag channels 
- **red:** base models trained with a mag channel and a phase channel


## Class 1: WM 

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/054756b0-b933-4fd0-9a3e-e67a4834cace" />

**Observation:** The mag only models (gray) have a slightly better score.

## Class 2: GM 

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/ea2cdd43-fe61-4ed1-937a-16c7102e4d67" />

**Observation:** No significant difference.

## Class 3: lesion on WM 

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/cdd47e92-6f1f-4726-96ff-e248086313d2" />

**Observation:** On the class lesions on WM, the mag alone is underperforming a bit.

## Class 4: lesion on GM 

<img width="510" height="431" alt="Image" src="https://github.com/user-attachments/assets/6664142a-a91e-4bb5-b915-7ad65f647a12" />

**Observation:** Lastly, on the class lesion on gray matter, the model trained with both the phase and the mag is significantly better. 

# Conclusion

The model using the mag only is only is slightly performing better on the WM segmentation. However, the results obtained with the base model are already good enough to question if they might be better than the ground truth themselves, hence this improvement is not really relevant. Moreover, the WM ground truth were made using the mag only, we could imagine the model using both the phase and the mag performs actually better than the gt because I rely on additional information from the phase. 
The models trained with mag and phase also perform better on the lesion related-classes, which are particularly challenging. **Therefore, we will keep working with the mag and phase as input for future trainings.**
